h/views/oidc.py

```diff
@@ -371,50 +384,57 @@ def redirect(self):
             raise HTTPForbidden
 
         # Get the user's provider unique ID from the provider.
-        provider_unique_id = self._oidc_service.get_provider_unique_id(
+        decoded_idtoken = self._oidc_service.get_decoded_idtoken(
             self.settings.provider, validated_params["code"]
         )
 
         # Get the existing Hypothesis account that's already connected to this
         # provider unique ID, if any.
         user = self._user_service.fetch_by_identity(
-            self.settings.provider, provider_unique_id
+            self.settings.provider, provider_unique_id=decoded_idtoken["sub"]
         )
 
         action = decoded_state.action
 
         match action:
             case "connect":
-                return self.connect_provider_unique_id(provider_unique_id, user)
+                return self.connect_identity(decoded_idtoken, user)
             case "login":
-                return self.log_in_with_provider(
-                    provider_unique_id, user, decoded_state.next_url
+                return self.log_in_with_identity(
+                    decoded_idtoken, user, decoded_state.next_url
                 )
             case _:
                 raise UnexpectedActionError(action)
 
-    def connect_provider_unique_id(self, provider_unique_id: str, user: User | None):
-        """Connect the user's provider unique ID to their Hypothesis account.
+    def connect_identity(self, decoded_idtoken: dict, user: User | None):
+        """Connect the user's third-party identity to their Hypothesis account.
 
-        Do nothing if `provider_unique_id` is already connected to `user`.
+        Do nothing if the third-party identity (represented by the given
+        `decoded_idtoken`) is already connected to `user`.
 
         """
         if user and user != self._request.user:
-            # Oops, this provider_unique_id is already connected to a different
+            # Oops, this identity is already connected to a different
             # Hypothesis account.
             raise UserConflictError
 
         if not user:
-            # This provider unique ID isn't connected to a Hypothesis account
-            # yet. Let's go ahead and connect it to the user's account.
+            # This identity isn't connected to a Hypothesis account yet. Let's
+            # go ahead and connect it to the user's account.
             self._oidc_service.add_identity(
-                self._request.user, self.settings.provider, provider_unique_id
+                self._request.user,
+                self.settings.provider,
+                provider_unique_id=decoded_idtoken["sub"],
+                email=decoded_idtoken.get("email"),
+                name=decoded_idtoken.get("name"),
+                given_name=decoded_idtoken.get("given_name"),
+                family_name=decoded_idtoken.get("family_name"),
```

Comment:
This is adding the new email and name-related fields to the `UserIdentity` table again, this time when connecting an identity to an existing account (rather than when creating a new account from an identity).